content/en/security/application_security/enabling/compatibility/nodejs.md

```diff
@@ -6,46 +6,163 @@ type: multi-code-lang
+| fs      |  {{< X >}} |  |  {{< X >}}  |
+| http    |  {{< X >}} |  {{< X >}}  |  {{< X >}}  |
+| https   |  {{< X >}} |  {{< X >}} |  |
+| http2   |  {{< X >}} |  {{< X >}} |  |
+
+
+<div class="alert alert-info">If you don't see your framework of choice listed, let us know! Fill out <a href="https://forms.gle/gHrxGQMEnAobukfn7">this short form to send details</a>.</div>
+
+### Data store compatibility
+
+
+**Datastore tracing provides:**
+
+- Timing request to response
+- Query info (for example, a sanitized query string)
+- Error and stacktrace capturing
+
+##### ASM Capability Notes
+- Open Source Vulnerability Detection is supported on all frameworks
+- Threat Protection works at the HTTP request (input) layer, and so works for all databases by default
+
+
+| Framework                 | Versions | Threat Detection supported? | Threat Protection supported? | Custom Code Vulnerability Detection supported? |
+| ----------------------- | ---------- | --------------- | ---------------------------------------------- | ---------------------------------------------- |
+| [cassandra-driver][28] | `>=3`    |{{< X >}}|{{< X >}} |          |
+| [couchbase][29]        | `^2.4.2` |{{< X >}}|          |          |
+| [elasticsearch][30]    | `>=10`   |{{< X >}}|          |          |
+| [ioredis][31]          | `>=2`    |{{< X >}}|{{< X >}} |          |
+| [knex][32]             | `>=0.8`  |{{< X >}}|{{< X >}} |          |
+| [mariadb][5]          | `>=3`    |{{< X >}}|{{< X >}} |          |
+| [memcached][33]        | `>=2.2`  |{{< X >}}|{{< X >}} |          |
+| [mongodb-core][34]     | `>=2`    |{{< X >}}|{{< X >}} |          |
+| [mysql][35]            | `>=2`    |{{< X >}}|          |{{< X >}} |
+| [mysql2][36]           | `>=1`    |{{< X >}}|{{< X >}} |{{< X >}} |
+| [oracledb][37]         | `>=5`    |{{< X >}}|{{< X >}} |          |
+| [pg][38]               | `>=4`    |{{< X >}}|{{< X >}} |{{< X >}} |
+| [redis][39]            | `>=0.12` |{{< X >}}|{{< X >}} |          |
+| [sharedb][40]          | `>=1`    |{{< X >}}|{{< X >}} |          |
+| [tedious][41]          | `>=1`    |{{< X >}}|{{< X >}} |          |
+
+
 [1]: /tracing/trace_collection/compatibility/nodejs/
 [2]: /agent/remote_config/#enabling-remote-configuration
+[4]: https://github.com/nodejs/release#release-schedule
+[5]: https://github.com/mariadb-corporation/mariadb-connector-nodejs
+[28]: https://github.com/datastax/nodejs-driver
+[29]: https://github.com/couchbase/couchnode
+[30]: https://github.com/elastic/elasticsearch-js
+[31]: https://github.com/luin/ioredis
+[32]: https://knexjs.org
+[33]: https://github.com/3rd-Eden/memcached
+[34]: http://mongodb.github.io/node-mongodb-native/core
+[35]: https://github.com/mysqljs/mysql
+[36]: https://github.com/sidorares/node-mysql2
+[37]: https://oracle.github.io/node-oracledb/
+[38]: https://node-postgres.com
+[39]: https://github.com/NodeRedis/node_redis
+[40]: https://share.github.io/sharedb/
+[41]: http://tediousjs.github.io/tedious
```

Comment:
This looks directly quoted from the APM documentation, some of it should be linked instead of copied, and a lot of it is just plain wrong because we have different framework support than APM

Reply:
> we have different framework support than APM

But AFAIK, it affects only to web frameworks, to detect a threat related with mariadb, mysql or other framework, we don't need to do anything. If apm support the framework, we will have an span with the query.
The only important thing is that the only web framework that we currently support is expressjs.

Reply:
For threat detection, the libraries like redis, pg, ... are not important, we don't have to do anything with this libraries if the tracer is already creating spans.
The important points are the web frameworks, of which currently we only support express.

Reply:
updated this section